xdsl/irdl.py

```diff
@@ -971,15 +971,15 @@ class VarIRConstruct(Enum):
 
 def get_construct_name(construct: VarIRConstruct) -> str:
     """Get the type name, this is used mostly for error messages."""
-    if construct == VarIRConstruct.OPERAND:
-        return "operand"
-    if construct == VarIRConstruct.RESULT:
-        return "result"
-    if construct == VarIRConstruct.REGION:
-        return "region"
-    if construct == VarIRConstruct.SUCCESSOR:
-        return "successor"
-    assert False, "Unknown VarIRConstruct value"
+    match construct:
```

Comment:
I would vote for the if instead of match.
Is there a reason for this?

Reply:
A match makes the check for exhaustiveness more explicit, and I would argue is the structure to use for matching on enums, or matching on a list of cases in general.

Reply:
Yeah match was designed for this use-case specifically it feels like to me